salt/_states/kubernetes.py

```diff
@@ -1009,6 +1009,171 @@ def node_label_present(
     return ret
 
 
+def node_annotation_absent(name, node, **kwargs):
+    '''
+    Ensures that the named annotation is absent from the node.
+
+    name
+        The name of the annotation
+
+    node
+        The name of the node
+    '''
+
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    annotations = __salt__['kubernetes.node_annotations'](node, **kwargs)
+
+    if name not in annotations:
+        ret['result'] = True if not __opts__['test'] else None
+        ret['comment'] = 'The annotation does not exist'
+        return ret
+
+    if __opts__['test']:
+        ret['comment'] = 'The annotation is going to be deleted'
+        ret['result'] = None
+        return ret
+
+    __salt__['kubernetes.node_remove_annotation'](
+        node_name=node,
+        annotation_name=name,
+        **kwargs
+    )
+
+    ret['result'] = True
+    ret['changes'] = {
+        'kubernetes.node_annotation': {'new': 'absent', 'old': 'present'}
+    }
+    ret['comment'] = 'Annotation removed from node'
+
+    return ret
+
+
+def node_annotation_present(
+        name,
+        node,
+        value,
+        **kwargs):
+    '''
+    Ensures that the named annotation is set on the named node
+    with the given value.
+    If the annotation exists it will be replaced.
+
+    name
+        The name of the annotation.
+
+    value
+        Value of the annotation.
+
+    node
+        Node to change.
+    '''
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    annotations = __salt__['kubernetes.node_annotations'](node, **kwargs)
+
+    if name not in annotations:
+        ret['comment'] = 'The annotation is going to be set'
+        if __opts__['test']:
+            ret['result'] = None
+            return ret
+
+        __salt__['kubernetes.node_add_annotation'](
+            annotation_name=name,
+            annotation_value=value,
+            node_name=node,
+            **kwargs
+        )
+
+    elif annotations[name] == value:
+        ret['result'] = True
+        ret['comment'] = (
+            'The annotation is already set and has the specified value'
+        )
+        return ret
+
+    else:
+        if __opts__['test']:
+            ret['result'] = None
+            ret['comment'] = 'The annotation is going to be updated'
+            return ret
+
+        ret['comment'] = 'The annotation is already set, changing the value'
+
+        __salt__['kubernetes.node_add_annotation'](
+            node_name=node,
+            annotation_name=name,
+            annotation_value=value,
+            **kwargs
+        )
+
+    old_annotations = copy.copy(annotations)
```

Comment:
Same comment as for the signature of `node_annotations`, I copy-pasted what was done for labels, and they indeed show the whole labels block instead of simply the one changed... To discuss in future improvements.